.openapidoc/index.js

```diff
@@ -0,0 +1,167 @@
+ */
+function cleanDistDir() {
+  fs.rmdirSync(distDir, { recursive: true });
+}
+
+/**
+ * Parse OpenApi yaml and return spec version.
+ */
+function calculateVersionFromSpec() {
+  let data = yaml.safeLoad(fs.readFileSync(spec, "utf8"));
+  return data.info.version;
+}
+
+/**
+ * Determine if its a release version from spec version. It contains -dev- as a result from our gradle build
+ * @param specVersion
+ */
+function isReleaseVersion(specVersion) {
+  return !specVersion.includes("-dev-");
+}
+
+/**
+ * Copy spec yaml to dist as as eth2signer-latest.yaml and optionally as eth2signer-specVer.yaml
+ * @param specVersion
+ */
+function copySpecToDist(specVersion) {
+  fs.copyFileSync(spec, `${distDir}/eth2signer-latest.yaml`);
+
+  if (isReleaseVersion(specVersion)) {
+    fs.copyFileSync(spec, `${distDir}/eth2signer-${specVersion}.yaml`);
+  }
+}
+
+/**
+ * Parse and update versions.json
+ * @param specVersion
+ * @param versionJsonString
+ */
+function updateVersionJson(specVersion, versionJsonString) {
+  var versions = JSON.parse(versionJsonString);
+  versions[specVersion] = { spec: specVersion, source: specVersion };
+  versions["stable"] = { spec: specVersion, source: specVersion };
+  fs.writeFileSync(versionsFile, JSON.stringify(versions, null, 1));
+}
+
+/**
+ * Download versions.json for release version from gh-pages and then pass result to updateVersionJson
+ * @param specVersion
+ */
+async function downloadAndUpdateVersionsFile(specVersion) {
+  return new Promise((resolve, reject) => {
+    if (!isReleaseVersion(specVersion)) {
+      console.log("Not a release version, bypassing updating versions.json");
+      return resolve(false);
+    }
+
+    return downloadAsString(versionsFileUrl)
+      .then((value) => {
+        try {
+          updateVersionJson(specVersion, value);
+          return resolve(true);
+        } catch (err) {
+          return reject(err);
+        }
+      })
+      .catch((err) => {
+        reject(err);
+      });
+  });
+}
+
+/**
+ * Publish dist folder to gh-pages branch
+ * @param specVersion
+ */
+async function publishToGHPages(specVersion) {
+  return new Promise((resolve, reject) => {
+    ghpages.publish(
+      distDir,
+      {
+        add: true,
+        branch: branch,
+        repo: repo,
+        user: {
+          name: "CircleCI Build",
+          email: "ci-build@consensys.net",
+        },
+        message: `[skip ci] OpenAPI Publish ${specVersion}`,
+      },
+      (err) => {
+        if (err) {
+          reject(err);
+        }
+        resolve(true);
+      }
+    );
+  });
+}
+
+async function run() {
+  cleanDistDir();
+  ensureDistDir();
+
+  let specVersion = calculateVersionFromSpec();
+  console.log("Spec Version: " + specVersion);
+
+  copySpecToDist(specVersion);
+  await downloadAndUpdateVersionsFile(specVersion);
+  await publishToGHPages(specVersion);
+
+  console.log("OpenAPI specs published to gh-pages successfully.");
+}
+
+run().catch((error) => {
```

Comment:
put the catch code inside of a try/catch block in the `run` function

Reply:
moved to try/catch block

Reply:
the last run() (which I've renamed to main()) cannot be part of try/catch because it is not part of an async method.